consensus/types/src/chain_spec.rs

```diff
@@ -712,6 +711,35 @@ impl ChainSpec {
         }
     }
 
+    /// Worst-case compressed length for a given payload of size n when using snappy.
+    ///
+    /// https://github.com/google/snappy/blob/32ded457c0b1fe78ceb8397632c416568d6714a0/snappy.cc#L218C1-L218C47
+    /// https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#max_compressed_len
+    fn max_compressed_len_snappy(n: usize) -> Option<usize> {
+        32_usize.checked_add(n)?.checked_add(n / 6)
```

Comment:
Rather than panic'ing (we have expects following this), why not just return usize::MAX if it were to overflow?

Reply:
I think silently failing returning a higher max value might not be safe here and could potentially lead to consensus issues.
I have added a test at the end of this file to check that there are no panics based on default mainnet values.